tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/contact/OpenPaasContactsConsumer.java

```diff
@@ -0,0 +1,164 @@
+                                    RabbitMQConfiguration commonRabbitMQConfiguration,
+                                    EmailAddressContactSearchEngine contactSearchEngine,
+                                    OpenPaasRestClient openPaasRestClient) {
+        this.receiverProvider = receiverProvider;
+        this.sender = sender;
+        this.commonRabbitMQConfiguration = commonRabbitMQConfiguration;
+        this.contactSearchEngine = contactSearchEngine;
+        this.openPaasRestClient = openPaasRestClient;
+    }
+
+    public void start() {
+        Flux.concat(
+                sender.declareExchange(ExchangeSpecification.exchange(EXCHANGE_NAME)
+                    .durable(DURABLE).type(BuiltinExchangeType.FANOUT.getType())),
+                sender.declareQueue(QueueSpecification
+                    .queue(QUEUE_NAME)
+                    .durable(DURABLE)),
+                sender.bind(BindingSpecification.binding()
+                    .exchange(EXCHANGE_NAME)
+                    .queue(QUEUE_NAME)
+                    .routingKey(EMPTY_ROUTING_KEY)))
+            .then()
+            .block();
+
+        consumeContactsDisposable = doConsumeContactMessages();
+    }
+
+    private Disposable doConsumeContactMessages() {
+        return delivery()
+            .subscribeOn(Schedulers.boundedElastic())
+            .subscribe(delivery ->
+                messageConsume(delivery, new String(delivery.getBody(), StandardCharsets.UTF_8)));
+    }
+
+    public Flux<AcknowledgableDelivery> delivery() {
+        return Flux.using(receiverProvider::createReceiver,
+            receiver -> receiver.consumeManualAck(QUEUE_NAME),
+            Receiver::close);
+    }
+
+    private void messageConsume(AcknowledgableDelivery ackDelivery, String messagePayload) {
+        Mono.just(messagePayload)
+            .<ContactAddedRabbitMqMessage>handle((message, sink) -> {
+                try {
+                    sink.next(objectMapper.readValue(message, ContactAddedRabbitMqMessage.class));
+                } catch (JsonProcessingException e) {
+                    sink.error(new RuntimeException(e));
+                }
+            })
+            .handle((msg, sink) -> {
+                handleMessage(msg).block();
+                ackDelivery.ack();
+            })
+            .onErrorResume(e -> {
+                LOGGER.error("Failed to consume OpenPaaS added contact message", e);
+                return Mono.empty();
+            }).subscribe();
+    }
+
+    private Mono<Void> handleMessage(ContactAddedRabbitMqMessage contactAddedMessage) {
+        LOGGER.info("Consumed jCard object message: {}", contactAddedMessage);
+
+        String openPaasOwnerId = contactAddedMessage.userId();
+        return openPaasRestClient.getUserById(openPaasOwnerId)
+            .map(OpenPaasUserResponse::preferredEmail)
+            .mapNotNull(ownerEmail -> {
```

Comment:
mapNotNull ? When can we have null? If we can produce null data producer code needs to be modified.